bin/node/cli/src/chain_spec.rs

```diff
@@ -181,8 +181,8 @@ fn pangolin_build_spec_genesis() -> pangolin_runtime::GenesisConfig {
 	}
 	impl Keys {
 		fn new(sr25519: &str, ed25519: &str) -> Self {
-			let sr25519 = array_bytes::hex2array_unchecked!(sr25519, 32);
-			let ed25519 = array_bytes::hex2array_unchecked!(ed25519, 32);
+			let sr25519 = array_bytes::hex2array_unchecked(sr25519);
```

Comment:
```
pub fn hex2array_unchecked<H, const N: usize>(hex: H) -> [u8; N]
where
	H: AsRef<str>,
{
	hex2bytes_unchecked(hex).try_into().unwrap()
}
```

I'm curious how compiler can infer `N` from `&str` whose size is unknown in compile time as I understand.

Reply:
Depend on the later usage (require 32).

Also `unchecked` means it won't check the `&str`'s length and always try to cast it to 32, may cause panic.